### ISSUE TITLE
process idea: medium-size features as github issues

### ISSUE THREAD

**Issue by 6a68** [mozilla/chronicle#90]
Here's a suggestion for how we can semi-organize our work/roadmap into github issues:

- capture huge features per milestone
- capture subfeatures as issues within the milestone
- the title of the issue is a user story (not necessarily in canonical "As a X I want Y so that Z" form, maybe just "Allow users to delete visits.")
- individual development tasks are just checkboxes within the description of the issue
- when a feature ships, we mark the issue as done.

Why do this?
- Keeps us focused on getting whole things done together
- Makes github issues a bit more readable

Why maybe not do this?
- I tend to over-rotate on process, this might not really be needed
- Merges/commits don't track directly to specific bugs
- Discussion within an issue might be scattered
- Maybe the real solution is to have a separate clear visualization of our plans, and let issues be as granular as they naturally turn out to be
- Unclear how to filter out bugs vs features (maybe labels work, though)

**For example:**

-----

**Milestone feature:** Save and retrieve visits

**Feature we want to add:** enable users to delete visits

**Github issue title:** Allow users to delete visits in the web UI

**Github issue description:** Some discussion to help define the task, for example, do we want to delete one or multiple visits at a time; do we want to remove records from the DB, or temporarily mark as deleted (to enable undo); other comments, either product-focused or implementation-focused, that clarify scope.

**Tasks defined in the issue**:
- [ ] initial UX input? (could be mocks, discussion, a few notes)
- [ ] add visit delete REST API endpoint
- [ ] add visit delete DB API function
- [ ] add front-end UI to enable deleting visits
- [ ] final review? UX or product?
- [ ] release the feature to production

**Comment by nchapman** [comment]
I think this is great. We've been doing this a bit, but I think formalizing it this way is inline with what my intentions have been for organizing in GitHub.

Just a few notes:

> when a feature ships, we mark the issue as done.

I think we specifically define shipping here as all subtasks have landed in master.

> Merges/commits don't track directly to specific bugs

I think this is mostly okay. In special situations where it doesn't make sense to land one part of the code without the other, we could collaborate on the branch.

> Discussion within an issue might be scattered

I think the key here is that we keep the issues as focused as possible. For example, if we're implementing Search as an overall feature, we should be able to break that down into smaller pieces pretty easily. Each of those pieces might have shared tasks in different parts of the app, but it'll make it easier to reason about and track our progress for the sprint. Let's be flexible and break things out if there's too much in one issue.

> Maybe the real solution is to have a separate clear visualization of our plans, and let issues be as granular as they naturally turn out to be

I think it's easier to rely on one tool and GitHub is working out well so far.  I think we'll always have a mix of granularity in our issues. Right now we're doing lots of big ticket items so we have more frequent unwieldiness.

> Unclear how to filter out bugs vs features (maybe labels work, though)

I think labels are perfect for this. Let's use them more generously.

---

**In conclusion**: let's do this starting right now!

**Comment by 6a68** [comment]
cool! I'll add this to CONTRIBUTING and open a PR

**Comment by 6a68** [comment]
Actually...maybe this should wait until @nchapman decides on the right tool to use (might be something outside github)

**Comment by 6a68** [comment]
hmm, closing in favor of using waffle.io
